### ISSUE TITLE
Build 'vsphere-iso' errored: error creating vm: host '' not found

### ISSUE THREAD

**Issue by mashiutz** [jetbrains-infra/packer-builder-vsphere#295]
Hello There :)
Im trying to build a new Windows template (using latest version - 2.3).
We have a Cluster with DRS Enabled, so im using only a Cluster in my Json.
If im specifying a Resource Pool, i am also getting the same error.
through vSphere web UI, i have made sure that the cluster do no reside in any folder.
I have seen a few people having trouble with this error as well, but nothing that can help me....
This is my json file : 

`{
    "builders": [
      {
        "type":                 "vsphere-iso",
  
        "vcenter_server":       "vcenter.domain.co.il",
        "insecure_connection":  "false",
        "username":             "administrator@vsphere.local",
        "password":             "adminpass",
        "datacenter":           "Datacenter-IT",
        "Cluster":              "IT-256GB",
        "datastore":            "VMFS_ISO",
        
        "communicator":         "winrm",
        "winrm_username":       "Superman",
        "winrm_password":       "SupermanPass",

        "vm_name":              "TEMPLATE-2019-TERM",
        "folder":               "WinTemplate-IT",
        "convert_to_template":  "true",
        "cpus":                 "2",
        "cpu_cores":            "1",
        "ram":                  "4096",
        "network_card":         "vmxnet3",
        "disk_controller_type": "lsilogic-sas",
	      "guest_os_type":        "windows9Server64Guest",
        "disk_thin_provisioned": false,
        "disk_size":            "71680",
	      "iso_paths": [
          "[VMFS_ISO] Windows-ISO/SW_DVD9_Win_Server_STD_CORE_2019_1809.1_64Bit_English_DC_STD_MLF_X22-02970.ISO",
          "[VMFS_ISO] VM Tools 10.1.0/vmtools/VMTools 10.1.0-windows.iso"
        ],
        "floppy_files": [
          "floppy/autounattend.xml",
          "floppy/setup.ps1",
          "floppy/vmtools.cmd"
        ]
        
      }
    ]
  }`

I would really appreciate if someone could help me with this!
Many thanks!

**Comment by danonh** [comment]
Can you confirm that superman@vsphere.local has a proper rights to create VMs on hosts within the cluster? Most people define cluster level permissions without propagate checkbox enabled.If you login to vsphere using superman@vsphere.local can you create VMs on IT-256GB cluster?

**Comment by mashiutz** [comment]
Yep, i have just logged in with that user and created a VM in that cluster.
Maybe i need special permissions when working with the vSphere API (which is what packer uses) ?

**Comment by mashiutz** [comment]
@mkuzmin you know something about this?

**Comment by danonh** [comment]
@mkuzmin will not respond as he is not involved in vsphere packer builder project anymore. Try to change value for insecure_connection to true, also remove cpu_cores value for testing purposes.

**Comment by danonh** [comment]
also if above will not solve your problem, could you try use builder from fallowing link? https://teamcity.jetbrains.com/viewLog.html?buildId=2529292&buildTypeId=PackerVSphere_Build&tab=artifacts

**Comment by mashiutz** [comment]
Thank you very much!
I'll update you.
are you involved with the project now?

**Comment by danonh** [comment]
I'm not involved with the project, but same as you I use vsphere-iso and vsphere-clones builders provided by jetbrains. I use it successfully on vSphere 6.7 although some updates for the builder would be advised. More info about @mkuzmin involvement into this project can be found here: https://github.com/hashicorp/packer/pull/7168

**Comment by mashiutz** [comment]
First, thank you for your help :)
I have tried your suggestions above, but I’m getting the same results.
Hope someone has more input on this.

**Comment by danonh** [comment]
Everything looks good with your template file. I would assume it's a vSphere permission issue, although you said you have checked that already. Can you give superman@vsphere.local full administrator rights to your vCenter server for testing purposes (if you work on production environment better don't do that)? If not, check permissions on WinTemplate-IT folder and on your vswitch/dvswitch, for superman@vsphere.local. I had similar problem yesterday when someone took my permissions off from my esxi host (I used it in non-clustered environment).
Take a look here for permissions settings: https://github.com/jetbrains-infra/packer-builder-vsphere/issues/97

**Comment by nslracer** [comment]
What gets people a lot times is when their cluster is in a folder structure. If it is then you need to provide the full path for example "FolderName/ClusterName". I use the same structure for the vm_folder field. Hope that helps.

**Comment by jasonwilliams14** [comment]
@nslracer I've been running into this issue today. 2 node esxi cluster, and am running vcenter as a VM on one of the nodes. Still cant seem to get passed the issue (running DRS BTW). Driving me a bit mad.
Could vcenter as VM on one of the nodes be an issue?
SO close, but cant seem to figure out this same error:

`vsphere-iso output will be in this color.

==> vsphere-iso: Creating VM...
Build 'vsphere-iso' errored: error creating vm: resource pool 'Arkham Asylum/Resources/deployments' not found

==> Some builds didn't complete successfully and had errors:
--> vsphere-iso: error creating vm: resource pool 'Arkham Asylum/Resources/deployments' not found`

**Comment by nslracer** [comment]
@jasonwilliams14 - I also run a 2 node ESXi cluster with vCenter as one of the VM's so I know for sure that's not your issue. 

I believe the issue might be the space in your resource names. The builder just does not handle spaces well in the datacenter, cluster, vm_folder and resource pool fields. If it's an option I would remove all spaces on your VMware object names (Cluster, Datacenter, folders, etc), update your code and try again.

**Comment by jasonwilliams14** [comment]
@nslracer I will give that a shot. Thx

**Comment by nslracer** [comment]
@jasonwilliams14 Any luck?

**Comment by jasonwilliams14** [comment]
@nslracer heya...was going to respond today. I did get it working. i cleaned up the naming convention with no spaces and that seemed to be the issue. Now I can pull ubuntu/debian and centos images just fine now with my setup. Much appreciated!

**Comment by nslracer** [comment]
@jasonwilliams14 Nice!!

**Comment by mashiutz** [comment]
Hi everyone Involved,
I know its been a while...
I'm still getting the same result:
`Build 'vsphere-iso' errored: error creating vm: host '' not found.`
My resource pool's does not have any spaces in their names, only dashes.
I'm using the latest version of Packer which is currently 1.5.4.
I know that the vsphere-iso builder was integrated into packer.exe. Still no luck.
Any take on this?
Should I file an issue on the official Packer GitHub?
Thanks again :)

**Comment by codefox42** [comment]
I can also reproduce this issue with Packer 1.5.4 and vSphere 6.7 (having a cluster name containing a dash in the middle).

**Comment by Yornik** [comment]
Can confirm on Packer 1.5.4 with a cluster name with 2 dashes

**Comment by codefox42** [comment]
Removing the dash from the cluster name did not change the behavior:

    Build 'vsphere-iso' errored: error creating vm: host '' not found

Probably some other aspect is causing this error.

**Comment by mashiutz** [comment]
Have anyone found a way around this issue?
Do you know if this will get fixed in the next version?

**Comment by nslracer** [comment]
See if the answer on this thread will help

https://github.com/jetbrains-infra/packer-builder-vsphere/issues/285#issuecomment-589819250

**Comment by pisaniej** [comment]
For future reference, if anyone here has a `/` in their `network`, you'll need to escape it with`%2f` ...semi-related
